falcon/asgi/app.py

```diff
@@ -1298,7 +1313,10 @@ async def _handle_exception(  # type: ignore[override]
             try:
                 kwargs = {}
 
-                if ws and 'ws' in get_argnames(err_handler):
+                # PERF(vytas): Using the LRU-cache backed misc._has_arg_name
+                #   here as inspect.signature (and by proxy misc.get_argnames)
+                #   can be very slow (on the order of magnitude of 10 µs).
+                if ws and _has_arg_name(err_handler, 'ws'):
```

Comment:
let's use `ws is not None`

Reply:
Hm, I would keep it _as is_ (and as it was before my perf fix unless you vehemently object.

This is a private method where we know that the incoming object is `WebSocket | None`, so these checks are functionally equivalent unless you somehow manage to monkey-patch the WS class or object to become falsy. And (counterintuitively I agree), `if obj` is slightly more performant than `if obj is None`. If `None` was implemented as `*PyObject == NULL`, it would be the opposite, but apparently it is not exactly like that in CPython.

Reply:
> `if obj` is slightly more performant than `if obj is None`

are you sure? I get the opposite result and I always knew it was faster

```
In [1]: class W:
   ...:     pass
   ...:

In [2]: w= W()

In [3]: %timeit if w: pass
16.8 ns ± 0.719 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)

In [4]: %timeit if w is not None: pass
14.2 ns ± 0.41 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
```